mlir/lib/Dialect/Func/Extensions/InlinerExtension.cpp

```diff
@@ -23,14 +23,16 @@ namespace {
 struct FuncInlinerInterface : public DialectInlinerInterface {
   using DialectInlinerInterface::DialectInlinerInterface;
 
+  static constexpr const char* NoInlineAttrName = "noinline";
```

Comment:
Hmm, would it make sense to have this as a proper ODS attribute on `FuncOp` and `CallOp`? LLVM dialect already has one https://github.com/llvm/llvm-project/blob/cc54a0ce5674b740c2136d7bd2416ffeb4a230cf/mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td#L1569.

Reply:
Yeah, it'd be good to integrate this directly into the dialect if possible (makes it much cleaner to target). Otherwise you're at the whims of attribute propagation.